File: android/app/src/main/java/com/opensource/xtsample/FooClass.kt
package com.opensource.ext.implementations

import com.eclipsesource.v8.V8Object
import com.opensource.xt.core.XTComponentExport
import com.opensource.xt.core.XTContext
import com.opensource.xt.core.XTExtObject

/**
 * Created by cuiminghui on 2018/1/27.
 */
class FooClass: XTExtObject.Implementation() {

    var fooValue: String = ""

    fun callYamiedie(roleA: String, roleB: String): String {
        invokeMethod("handleNativeCall", listOf())
        return "$roleB said: '$roleA Yamiedie'."
    }

}

File: src/Sample/core/FooClass.ts
if (XT.platform === "Web") {

    (window as any)["FooClass"] = class FooClass {

        fooValue: string = ""

        invokeMethod: any;

        callYamiedie(roleA: string, roleB: string) {
            this.invokeMethod("handleNativeCall", [])
            return roleB + " said: '" + roleA + " Yamiedie'."
        }

    }
    
}